README.md

```diff
@@ -0,0 +1,159 @@
+# device-coap-c
+
+EdgeX device service for CoAP-based REST protocol
+
+This device service allows a 3rd party sensor application to push data into EdgeX via CoAP. Like HTTP, CoAP provides REST based access to resources, but CoAP is more compact for use in constrained IoT devices.
+
+The device-coap-c service (_device-coap_ for short) is modeled after the HTTP based [device-rest-go](https://github.com/edgexfoundry/device-rest-go) service, and runs over UDP. The current implementation is meant for one-way communication from a device into EdgeX by posting readings asynchronously. Synchronous interaction initiated by EdgeX can be added in the future.
+
+device-coap uses DTLS for secure communication to devices. It is written in C, and relies on the well known [libcoap](https://libcoap.net/) library.
+
+For background on device-coap-c, CoAP and low power wireless, see the [presentation](https://zoom.us/rec/share/N2Uh7C9qScsj32bs0T8aNF4VPPOuFSypnhQp3g2LmSFfOA16giRq9gwqpGvNb1HX.kknLyNV7Rj72mPms?startTime=1602514686000) to the Device Working Group.
+
+## Resources
+
+device-coap creates a parameterized CoAP resource to which data may be posted:
+
+```
+   /a1r/{deviceName}/{resourceName}
+```
+
+- `a1r` is short for "API v1 resource", as defined by device-rest-go.
+- `deviceName` refers to a `device` managed by the CoAP device service. For example, `res/configuration.toml` pre-defines a device named 'd1'.
+- `resourceName` refers to a `deviceResource` defined in the device profile, as described in the sub-section below.
+
+Payload data posted to one of these resources is type validated, and the resulting value then is sent into EdgeX via the Device SDK's asynchronous `post_readings` capability.
+
+## Profiles
+
+[example-datatype.yaml](./res/example-datatype.yaml) defines  generic resources for data types. The table below shows the available resource names and correspondence with CoAP attributes. 
+
+For example, the 'int' resource name means that EdgeX provides a CoAP resource, `/a1r/{deviceName}/int`. This resource accepts an integer encoded as text, like `42`.
+
+| resourceName | Type   | EdgeX MediaType<br>CoAP Content-Format|
+|---------|--------|---------------------------------------|
+| int     | Int32  | text/plain                            |
+| float   | Float64| text/plain                            |
+| json    | String | application/json                      |
+
+>_Note:_ You must define the Content-Format option in the CoAP POST request. See the _Testing_ section below for example use.
+
+
+## Configuration
+
+This section describes properties in [configuration.toml](./res/configuration.toml) as used by device-coap. See the _Configuration and Registry_ section of the EdgeX documentation for background.
+
+### Driver
+
+Below are the recognized properties for the Driver section, followed by an example.
```

Comment:
Should note that these are not dynamic: the device service must be restarted for changes to take effect

Reply:
Addressed in 9fa2601